core/imds/src/test/java/software/amazon/awssdk/imds/Ec2MetadataTest.java

```diff
@@ -40,4 +40,36 @@ public void when_dummy_string_is_returned(){
         assertThat(ec2Metadata.get("/ami-id")).isEqualTo("IMDS");
     }
 
+    @Test
+    public void when_default_retries_is_returned(){
+
+        Ec2Metadata client = Ec2Metadata.builder().build();
+        assertThat(client.retries()).isEqualTo(3);
+    }
+
+    @Test
+    public void when_custom_retries_is_returned(){
+
+        Ec2Metadata client = Ec2Metadata.builder().retries(4).build();
+        assertThat(client.retries()).isEqualTo(4);
+    }
+
+    @Test
+    public void verify_not_equal_objects(){
+
+        Ec2Metadata client = Ec2Metadata.builder().retries(2).build();
+        Ec2Metadata anotherClient = Ec2Metadata.builder().build();
+        assertThat(client.retries()).isNotEqualTo(anotherClient.retries());
+    }
+
+    @Test
+    public void verify_hashcode_equal(){
+
+        Ec2Metadata client = Ec2Metadata.builder().build();
+        Ec2Metadata anotherClient = Ec2Metadata.builder().build();
+        assertThat(client.hashCode()).isEqualTo(anotherClient.hashCode());
+    }
+
```

Comment:
Use `equalsVerifier` https://github.com/aws/aws-sdk-java-v2/blob/1032916a030735685f2977a806dc70304b87b77d/services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/internal/DefaultCopyTest.java#L25

Reply:
Did not know about it. Will implement